internal/resources/free_tier_bucket.go

```diff
@@ -0,0 +1,326 @@
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+}
+
+func (f *FreeTierBucket) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	var currState providerschema.Bucket
+	diags := request.State.Get(ctx, &currState)
+	response.Diagnostics.Append(diags...)
+
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	IDs, err := currState.Validate()
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Error reading free-tier bucket in Capella",
+			"Could not read free-tier Bucket with ID "+currState.Id.String()+": "+err.Error(),
+		)
+		return
+	}
+
+	var (
+		organizationId = IDs[providerschema.OrganizationId]
+		projectId      = IDs[providerschema.ProjectId]
+		clusterId      = IDs[providerschema.ClusterId]
+		bucketId       = IDs[providerschema.Id]
+	)
+
+	refreshedState, err := f.retrieveFreeTierBucket(ctx, organizationId, projectId, clusterId, bucketId)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Error reading free-tier bucket",
+			"could not read the free-tier bucket with ID "+currState.Id.String()+": "+err.Error(),
+		)
+		return
+	}
+
+	diags = response.State.Set(ctx, refreshedState)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (f *FreeTierBucket) Update(ctx context.Context, request resource.UpdateRequest, response *resource.UpdateResponse) {
+	var plannedState providerschema.Bucket
+	diags := request.Plan.Get(ctx, &plannedState)
+	response.Diagnostics.Append(diags...)
+
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	var currState providerschema.Bucket
+	diags = request.State.Get(ctx, &currState)
+	response.Diagnostics.Append(diags...)
+
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	organizationId := currState.OrganizationId.ValueString()
+	projectId := currState.ProjectId.ValueString()
+	clusterId := currState.ClusterId.ValueString()
+	bucketId := currState.Id.ValueString()
+
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/buckets/freeTier/%s", f.HostURL, organizationId, projectId, clusterId, bucketId)
+
+	updateBucketRequest := api.UpdateFreeTierBucketRequest{
+		MemoryAllocationInMb: plannedState.MemoryAllocationInMB.ValueInt64(),
+	}
+
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPut, SuccessStatus: http.StatusNoContent}
+	_, err := f.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		updateBucketRequest,
+		f.Token,
+		nil,
+	)
+	if err != nil {
+		resourceNotFound, errString := api.CheckResourceNotFoundError(err)
+		if resourceNotFound {
+			tflog.Info(ctx, "resource doesn't exist in remote server removing resource from state file")
+			response.State.RemoveResource(ctx)
+			return
+		}
+		response.Diagnostics.AddError(
+			"Error updating free-tier bucket",
+			"Could not update free-tier bucket, unexpected error: "+bucketId+": "+errString,
+		)
+		return
+	}
+
+	// Sleep for 5 seconds for the bucket stats to be updated for GET request
+	time.Sleep(time.Second * 5)
```

Comment:
why wait 5 sec ?  

shouldn't update take effect right away ?

Reply:
5 seconds seems excessive.  client has to wait 5 seconds. 

doesn't really matter if bucket stats are not updated.  they will get stale anyway.  

as long as bucket quota (`MemoryAllocationInMb`) is updated that's all we need.